io-engine/src/subsys/nvmf/target.rs

```diff
@@ -355,13 +350,20 @@ impl Target {
 
     /// Final state for the target during init.
     pub fn running(&mut self) {
-        self.enable_discovery();
-        info!(
-            "nvmf target accepting new connections and is ready to roll..{}",
-            '\u{1F483}'
-        );
+        let discovery = self.enable_discovery();
```

Comment:
With this change, `enable_discovery` is not really starting the discovery subsystem so perhaps rename the function to `get_discovery_subsystem` or `create_discovery_subsystem`

Reply:
Yep, done